api/v1beta2/bucket_types.go

```diff
@@ -100,6 +100,11 @@ type BucketSpec struct {
 	// +optional
 	CertSecretRef *meta.LocalObjectReference `json:"certSecretRef,omitempty"`
 
+	// ProxySecretRef specifies the Secret containing the proxy configuration
+	// to use while communicating with the Bucket server.
+	// +optional
```

Comment:
Can we specify here and in the API docs that this field is taking into consideration only for `provider: generic`?

Reply:
We will support all providers

Reply:
Well, it turns out we can't support GCS:

https://github.com/fluxcd/source-controller/pull/1520#issuecomment-2183523662

Reply:
> Can we specify here and in the API docs that this field is taking into consideration only for provider: generic?

fixed, thanks!